src/BizHawk.Client.EmuHawk/tools/ToolManager.cs

```diff
@@ -473,6 +473,19 @@ public bool Has(Type toolType)
 			return Load<T>(false);
 		}
 
+		public T FirstOrNull<T>(Predicate<T> condition) where T : class

```

Comment:
Type-check + `Predicate` seems weird. The caller should include `t is T &&` if that's desired.

Reply:
I agree this currently feels weird, but the function was specifically tailored to fit the needs of the callsite while staying somewhat generic. Having to cast twice, once in the predicate and once for casting the return value felt weird also ( like `(IControlMainform)Tools.FirstOrNull(static t => t is IControlMainform tool && tool.WantsToControlRewind);`).

Reply:
You could also make `ToolManager` implement `IEnumerable<IToolForm>` and use regular LINQ at the callsites.